### DISCUSSION TITLE
Implementing a custom hit shader for OptiX

### DISCUSSION THREAD

**Discussion by PLangsteiner** [shader-slang/slang#5484] [Q&A] [answered]
Hello Everyone,

I am currently trying to implement my shaders in Slang using OptiX as my rendering engine. 

Currently, I am trying to implement a sphere intersection, and it works fine when I write it in CUDA, since I can use the "OptixReportIntersection" function there. 
However, I cannot find anything similar to this function in Slang.
Does anyone know if there is a similar built-in function, that reports an intersection back to OptiX, or any other way?

Thank you for your help in advance.

**Comment by csyonghe** [comment, marked as answer]
Please call `ReportHitOptix` in Slang: https://shader-slang.com/stdlib-reference/global-decls/reporthitoptix-069
